Adds command "tabmove ±n", for better tab management

jwarner112 opened the pull request:
I'm used to being able to switch and swap tabs all the time and couldn't find a way to do it in here, yet. I haven't made this a default for anything (was thinking `Alt-<` and `Alt->` could do left/right) ~~and haven't put it in any documentation, yet~~ (now I have!)

Some usage info I _did_ type up, though:

----

Tab Move
---------

Moves active tab to another slot while maintaining its focus.

### Usage

`> tabmove ±?n`

1. `±` is an *optional* character, either `+` or `-`
	- if it prefixes `n`, `n` will describe a relative position (left and right, respectively)
2. `n` is an integer.
	- if not prefixed by `±`, describes an absolute position (resolving to the closest slot available)

### Examples

`> tabmove 20`	moves active tab to slot `min(numSlots, 20)`
`> tabmove 3` 	moves active tab to slot `min(numSlots, 3)`
`> tabmove 0` 	moves active tab to slot `0`
`> tabmove -2`  moves active tab left by (at most) `2` 
`> tabmove +1`  moves active tab right by (at most) `1`
`> tabmove +20` moves active tab right by (at most) `20`

### Notes

I'm used to this command in Vim and VSCode, but it's missing from `micro`, so I added it.

### Action Items	

- [x] Get command working
- [x] Custom keybinding for `command:tabmove ±1`
- [x] Documentation

zyedidia commented:
I think it would be better to just add this new relative syntax to the `tabswitch` command rather than make a new `tabmove` command. Also, `Alt-,` and `Alt-.` (`Alt-<`, and `Alt->` without shift) already move one tab to the left or right.

jwarner112 commented:
Ah, I'm afraid I've been unclear. The existing command `tabswitch` moves the *focus* from one tab to another:
![tabswitch](https://user-images.githubusercontent.com/2992409/81020547-e8d9b900-8e1d-11ea-9f0c-3869cb7fefc3.gif)

The command I've added, `tabmove`, moves *the tab itself*:
![tabmove](https://user-images.githubusercontent.com/2992409/81020660-26d6dd00-8e1e-11ea-9c18-e4240e469546.gif)

On the note of adding the relative syntax to `tabswitch`: I did want to do that in an upcoming PR but I had intended to hold off on that until I spoke to you about how commands and actions should relate to one another.

TLDR; `tabswitch` overlaps responsibilities with `NextTab` and `PreviousTab`. At the moment, the keybindings match directly to `actions.NextTab` and `actions.PreviousTab`, where once the relative syntax is implemented the keybinding can instead map to `command:tabswitch -1` and `command:tabswitch +1`, respectively.

If we moved forward with this style of keybinding->command, we'd be approaching the Vim style of editing/configuration, wherein *everything* is just a command. However, if we go in the opposite direction, then every action that keybindings meet with will *probably* have a command that does the same thing. At the moment, if there's a way to specifically have actions call commands, I just haven't figured it out yet (but haven't put much time into it, either)

jwarner112 commented:
@zyedidia Hey there, just curious if you'd seen my elaboration and if it changed your mind at all?

zyedidia commented:
We can also think about having click and drag work to move tabs around.
